pom.xml

```diff
@@ -370,6 +370,49 @@
                         </excludes>
                     </configuration>
                 </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-enforcer-plugin</artifactId>
+                    <executions>
+                        <execution>
+                            <id>banned-commons-logging</id>
+                            <goals>
+                                <goal>enforce</goal>
+                            </goals>
+                            <configuration>
+                                <rules>
+                                    <bannedDependencies>
+                                        <excludes>
+                                            <exclude>commons-logging:commons-logging</exclude>
+                                        </excludes>
+                                        <message>We don't use commons-logging any more, so do not depend on it directly.</message>
+                                        <searchTransitive>false</searchTransitive>
+                                    </bannedDependencies>
+                                </rules>
+                            </configuration>
+                        </execution>
+                        <execution>
+                            <id>banned-other-logging-framework</id>
+                            <goals>
+                                <goal>enforce</goal>
+                            </goals>
+                            <configuration>
+                                <rules>
+                                    <bannedDependencies>
+                                        <excludes>
+                                            <exclude>log4j:*</exclude>
```

Comment:
Do we even need to allow log4j2 ?
Isn't all logging happening via slf4j ?

Reply:
We use log4j2. slf4j is just a facade/abstraction layer and doesnot perform any logging in itself. Log4j2 on other hand provides actual logging implementation. So we do need both innthe project.

Reply:
I thought that these rules only apply to the source code references, and not to the transitive dependencies.
But I have never tested.

Reply:
Ah we could just restric imports via https://github.com/skuzzle/restrict-imports-enforcer-rule. But maven-enforcer-plugin will ban transitive dependecies as well.

Reply:
Thanks for the explanation.